experimental/PlaneCurve/ProjCurve.jl

```diff
@@ -21,6 +50,222 @@ end
+, MPolyElem_dec{fmpq, fmpq_mpoly}[x[0], x[1], x[2], x[3]])
+
+julia> P = Oscar.Geometry.ProjSpcElem(PP[1], [QQ(0), QQ(2), QQ(0), QQ(5)])
+(0 : 2 : 0 : 5)
+
+julia> P in C
+true
+```
+"""
+function Base.in(P::Oscar.Geometry.ProjSpcElem, C::ProjCurve)
+    I = Oscar.defining_ideal(C)
+    V = I.gens.O
+    s = length(V)
+    i = 1
+    while i <= s
+        if !iszero(evaluate(V[i], P.v))
+            return false
+        end
+        i = i + 1
+    end
+    return true
+end
+
+################################################################################
+@doc Markdown.doc"""
+    curve_components(C::ProjCurve)
+
+Return a dictionary containing the irreducible components of `C` and the
+corresponding reduced curve.
+"""
+function curve_components(C::ProjCurve)
+    if isempty(C.components)
+        I = Oscar.defining_ideal(C)
+        L = primary_decomposition(I)
+        s = length(L)
+        D = Dict{ProjCurve, ProjCurve}()
+        for i in 1:s
+            if dim(L[i][1]) == 2
+                D[ProjCurve(L[i][1])] = ProjCurve(L[i][2])
+            end
+        end
+        C.components = D
+    end
+    return C.components
+end
+
+################################################################################
+
+@doc Markdown.doc"""
+    isirreducible(C::ProjCurve)
+
+Return `true` if `C` is irreducible, and `false` otherwise.
+
+# Example
+```jldoctest
+julia> S, (x, y, z, t) = PolynomialRing(QQ, ["x", "y", "z", "t"])
+(Multivariate Polynomial Ring in x, y, z, t over Rational Field, fmpq_mpoly[x, y, z, t])
+
+julia> T, _ = grade(S)
+(Multivariate Polynomial Ring in x, y, z, t over Rational Field graded by
+  x -> [1]
+  y -> [1]
+  z -> [1]
+  t -> [1], MPolyElem_dec{fmpq, fmpq_mpoly}[x, y, z, t])
+
+julia> I = ideal(T, [x^2, y^2*z, z^2])
+ideal(x^2, y^2*z, z^2)
+
+julia> C = Oscar.ProjCurve(I)
+Projective curve defined by the ideal(x^2, y^2*z, z^2)
+
+julia> Oscar.isirreducible(C)
+true
+```
+"""
+function Oscar.isirreducible(C::ProjCurve)
+    L = curve_components(C)
+    return length(L) == 1
+end
+
+################################################################################
+
+@doc Markdown.doc"""
+    reduction(C::ProjCurve)
+
+Return the projective curve defined by the radical of the defining ideal of `C`.
+
+# Example
+```jldoctest
+julia> S, (x, y, z, t) = PolynomialRing(QQ, ["x", "y", "z", "t"])
+(Multivariate Polynomial Ring in x, y, z, t over Rational Field, fmpq_mpoly[x, y, z, t])
+
+julia> T, _ = grade(S)
+(Multivariate Polynomial Ring in x, y, z, t over Rational Field graded by
+  x -> [1]
+  y -> [1]
+  z -> [1]
+  t -> [1], MPolyElem_dec{fmpq, fmpq_mpoly}[x, y, z, t])
+
+julia> I = ideal(T, [x^2, y^2*z, z^2])
+ideal(x^2, y^2*z, z^2)
+
+julia> C = Oscar.ProjCurve(I)
+Projective curve defined by the ideal(x^2, y^2*z, z^2)
+
+julia> Oscar.reduction(C)
+Projective curve defined by the ideal(z, x)
+```
+"""
+function reduction(C::ProjCurve)
+    J = radical(Oscar.defining_ideal(C))
```

Comment:
I think it would be nicer to once write `import Oscar.defining_ideal` at the start of your module(s), and then replace `Oscar.defining_ideal` by `defining_ideal` everywhere else.